compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -1149,8 +1149,12 @@ pub enum CastKind {
     Pointer(PointerCast),
     /// Cast into a dyn* object.
     DynStar,
-    /// Remaining unclassified casts.
-    Misc,
+    IntToInt,
+    FloatToInt,
+    FloatToFloat,
+    IntToFloat,
+    PtrToPtr,
```

Comment:
Non-blocking request from the peanut gallery: It's not obvious to me how `PtrToPtr` differs from things in `Pointer(_)`.  Could you please put some doc-comments on the new variants?

Reply:
"FIXME: `Pointer(_)` is legacy code and should be removed. Please submit a PR if you feel inclined." :upside_down_face:

Reply:
I haven't investigated the behavior of `Pointer(_)` but old code did it like the below, and some `Ptr` to `Ptr` casts went through `CastKind::Misc` e.g code in `src/test/ui/simd/intrinsic/ptr-cast.rs`. However, I could also investigate what `Pointer(_)` is actually doing in the next PR.

```rust
(_, _) => CastKind::Misc,
```